clusters/resource_cluster.go

```diff
@@ -156,93 +211,110 @@ func resourceClusterSchema() map[string]*schema.Schema {
 }
 
 func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-	var cluster Cluster
 	start := time.Now()
 	timeout := d.Timeout(schema.TimeoutCreate)
-	clusters := NewClustersAPI(ctx, c)
+	w, err := c.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+	clusters := w.Clusters
+	var cluster compute.CreateCluster
 	common.DataToStructPointer(d, clusterSchema, &cluster)
-	if err := cluster.Validate(); err != nil {
+	if err := Validate(cluster); err != nil {
```

Comment:
Not your issue, but shouldn't this be in CustomizeDiff? Cluster spec should be validated there so we fail before applying.

Reply:
Yes you are right, it's has been marked as a TODO since long time back: https://github.com/databricks/terraform-provider-databricks/blob/main/clusters/clusters_api.go#L437. I have created a followup ticket to do this so we don't risk introducing behaviour changes in this PR.